packages/stardog-graphql-language-server/src/GraphQlLanguageServer.ts

```diff
@@ -70,6 +91,111 @@ export class GraphQlLanguageServer extends AbstractLanguageServer<
+        this.namespaceMap,
+        this.relationshipBindings.map((binding) => ({
+          iri: binding.relationship.value,
+          count: binding.count.value,
+        })),
+        CompletionItemKind.Field
+      );
+    }
+    if (update.typeBindings || (update.namespaces && this.typeBindings)) {
+      this.typeBindings = update.typeBindings || this.typeBindings;
+      this.typeCompletionItems = this.buildCompletionItemsFromData(
+        this.namespaceMap,
+        this.typeBindings.map((binding) => ({
+          iri: binding.type.value,
+          count: binding.count.value,
+        })),
+        CompletionItemKind.EnumMember
+      );
+    }
+    if (
+      update.graphQLValueTypeBindings ||
+      (update.namespaces && this.graphQLValueTypeBindings)
+    ) {
+      this.graphQLValueTypeBindings =
+        update.graphQLValueTypeBindings || this.graphQLValueTypeBindings;
+      this.graphQLTypeCompletionItems = this.buildGraphQLTypeCompletionItems(
+        this.namespaceMap,
+        this.graphQLValueTypeBindings
+      );
+    }
+  }
+
+  buildCompletionItemsFromData(
+    namespaceMap,
+    irisAndCounts: { iri: string; count: string }[],
+    kind: CompletionItemKind
+  ): CompletionItem[] {
+    // GraphQL cannot idenfy non-prefixed IRIs as fields, so we want to ignore non-prefixed bindings
+    const prefixed: CompletionItem[] = [];
+
+    if (!namespaceMap) {
+      return prefixed;
+    }
+
+    irisAndCounts.forEach(({ iri, count }) => {
+      const prefixedIri = abbreviatePrefixObj(iri, namespaceMap);
+      if (prefixedIri !== iri) {
+        // in GraphQL, the `:` character cannot be used in field names so instead Stardog uses the `_` character
+        const graphQLIri = prefixedIri.replace(/\:/g, '_');
+        const alphaSortTextForCount =
+          ARBITRARILY_LARGE_NUMBER - parseInt(count, 10);
+        prefixed.push({
+          label: graphQLIri.slice(graphQLIri.startsWith('_') ? 1 : 0),
+          kind,
+
+          // take the difference of an arbitrarily large number and the iri's count which
+          // allows us to invert the sort order of the items to be highest count number first
+          sortText: `${kind}-${alphaSortTextForCount}${graphQLIri}`,
+
+          // concatenate both the full iri and the prefixed iri so that users who begin
+          // typing the full or prefixed iri will see the graphql alternative
+          filterText: `${iri}${prefixedIri}${graphQLIri}`,
+          detail: `${count} occurrences`,
+        });
+      }
+    });
+
+    return prefixed;
+  }
+
+  buildGraphQLTypeCompletionItems(
+    namespacesMap,
+    typesAndIris: { type: string; iri: string }[]
+  ): CompletionItem[] {
+    return GRAPHQL_VALUE_TYPES.map((graphQLType) => {
+      const completionItem: CompletionItem = {
+        label: graphQLType,
+        kind: CompletionItemKind.Enum,
+      };
+      const typeAndIri = typesAndIris.find((t) => t.type === graphQLType);
+      if (typeAndIri && namespacesMap) {
```

Comment:
Again, I don't think `namespacesMap` is correct, for reasons mentioned above. Not sure it's ever falsey.

Also, I'm noting here that it is sometimes called `namespacesMap` (plural "namespaces") and sometimes `namespaceMap` (singular "namespace").